src/routes/landing/components/CityEntry.js

```diff
@@ -62,13 +64,13 @@ class CityEntry extends React.PureComponent<PropType> {
                     underlayColor={theme.colors.backgroundAccentColor}>
         <View>
           <Label theme={theme} searchWords={[filterText]} textToHighlight={city.name} sanitize={normalizeSearchString}
-                 highlightStyle={{ backgroundColor: theme.colors.themeColor }} />
+                 highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }} />
           {aliases.length > 0 && <Aliases>
-            {aliases.map(
+            {aliases.slice(0, MAX_NUMBER_OF_ALIASES_SHOWN).map(
               (alias, index) => <AliasLabel key={alias} theme={theme} searchWords={[filterText]}
-                                            textToHighlight={index === aliases.length - 1 ? alias : `${alias}, `}
+                                            textToHighlight={index !== aliases.slice(0, MAX_NUMBER_OF_ALIASES_SHOWN).length - 1 ? `${alias}, ` : aliases.length <= MAX_NUMBER_OF_ALIASES_SHOWN ? alias : `${alias},...`}
```

Comment:
Line too long and unclear. Also, afaics, the comma and dots (`,...`) are also part of an alias, that should not be the case. Also, a space is missing between the comma and the dots and this way it won't work for RTL layouts (as it is always `, ...` and never `... ,` (same for aliases) as it should be for RTL layouts. Therefore, you'd need to split `alias`, `,` and `...`. Maybe also extract some logic here as it gets quite unclear.